packages/compass-home/src/components/home.tsx

```diff
@@ -143,6 +152,25 @@ function notifyMainProcessOfDisconnect() {
   void ipcRenderer?.call('compass:disconnected');
 }
 
+export const CompassConnectionProviderContext: React.FunctionComponent<{}> = ({
+  children,
+}) => {
+  const storage = useContext(ConnectionStorageContext);
+  if (!storage) {
+    throw new Error(
+      'Could not find current ConnectionStorage. Did you forget to setup the ConnectionStorageContext?'
+    );
+  }
+
+  return (
+    <ConnectionProviderContext.Provider
+      value={new CompassConnectionProvider(storage)}
```

Comment:
Not sure if we're running the risk of accidental re-renders here in some form, maybe this should be a `const connectionProvider = useMemo(() => new CompassConnectionProvider(storage), [storage]);` so that we don't create new `CompassConnectionProvider` objects whenever this component re-renders (which I'm not entirely sure happens all that frequently, so idk if this makes sense)?

Reply:
Well the storage is a singleton, so it shouldn't happen, but let me change it `just in case` we change it in the future.